lib/plivo/xml/multipartycall.rb

```diff
@@ -0,0 +1,175 @@
+module Plivo
+  module XML
+    class MultiPartyCall < Element
+      @nestables = []
+      @valid_attributes = %w[role maxDuration maxParticipants waitMusicUrl
+                             waitMusicMethod agentHoldMusicUrl agentHoldMusicMethod
+                             customerHoldMusicUrl customerHoldMusicMethod record
+                             recordFileFormat recordingCallbackUrl recordingCallbackMethod
+                             statusCallbackEvents statusCallbackUrl statusCallbackMethod
+                             stayAlone coachMode mute hold startMpcOnEnter endMpcOnExit
+                             enterSound enterSoundMethod exitSound exitSoundMethod
+                             onExitActionUrl onExitActionMethod relayDTMFInputs]
+
+      VALID_ROLE_VALUES = %w[agent supervisor customer]
+      VALID_MAX_DURATION_VALUES = (300..28800).to_a
```

Comment:
this would be an array of 28,501 elements
array search are of O(log N) complexity at best

since we know the use case here can't we simplify the check to be 
300 <  attributes[:maxDuration] < 28800